### PR TITLE
Add CoreDNS as an optional addon in kube-up

### PR DESCRIPTION
**Pull request by rajansandeep**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: cluster/addons/dns/coredns.yaml.in [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  name: kube-dns
+  namespace: kube-system
+  labels:
+    k8s-app: kube-dns
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+    kubernetes.io/name: "CoreDNS"
+spec:
+  selector:
+    k8s-app: kube-dns
```
Comment (chrisohaver) [inline cluster/addons/dns/coredns.yaml.in]: The selector should be `k8s-app: coredns`, so it matches the pod labels.

File: cluster/addons/dns/coredns.yaml.base [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  name: kube-dns
+  namespace: kube-system
+  labels:
+    k8s-app: kube-dns
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+    kubernetes.io/name: "CoreDNS"
+spec:
+  selector:
+    k8s-app: kube-dns
```
Comment (chrisohaver) [inline cluster/addons/dns/coredns.yaml.base]: The selector should be k8s-app: coredns, so it matches the pod labels...

File: cluster/addons/dns/coredns.yaml.sed [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  name: kube-dns
+  namespace: kube-system
+  labels:
+    k8s-app: kube-dns
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+    kubernetes.io/name: "CoreDNS"
+spec:
+  selector:
+    k8s-app: kube-dns
```
Comment (chrisohaver) [inline cluster/addons/dns/coredns.yaml.sed]: The selector should be k8s-app: coredns, so it matches the pod labels...

**Comment by fturib** [comment]
I am ok with the logic.
Thanks for the change !

**Pull request closed without merging**